dluhc-component-library/src/components/maps/MapComponent.tsx

```diff
@@ -47,6 +52,15 @@ const MapComponent = ({
     circleFillColor: "#ffcc33",
   },
 }: MapComponentProps) => {
+  const [datasets, setDatasets] = useState<string[]>([]);
+  const selectDataset = (dataset: string) => {
+    if (!datasets.includes(dataset)) {
+      setDatasets([...datasets, dataset]);
+    } else {
+      setDatasets(datasets.filter((d) => d !== dataset));
```

Comment:
We should use full variable names where possible. I know it is fairly obvious what `d` is when we read the line but at a scan you have to stop and think. In a previous file I know we have `a` and `b` but that's a common notation for comparable functions.